rnd/autogpt_server/autogpt_server/server/server.py

```diff
@@ -351,17 +361,69 @@ def execute_graph_block(
 
     @classmethod
     async def get_graphs(cls) -> list[str]:
-        return await get_graph_ids()
+        return await autogpt_server.data.graph.get_graph_ids()
+
+    @classmethod
+    async def get_templates(cls) -> list[autogpt_server.data.graph.GraphMeta]:
+        return await autogpt_server.data.graph.get_template_meta()
+
+    @classmethod
+    async def get_graph(cls, graph_id: str) -> autogpt_server.data.graph.Graph:
+        graph = await autogpt_server.data.graph.get_graph(graph_id)
+        if not graph:
+            raise HTTPException(status_code=404, detail=f"Graph #{graph_id} not found.")
+        return graph
 
     @classmethod
-    async def get_graph(cls, graph_id: str) -> Graph:
-        graph = await get_graph(graph_id)
+    async def get_template(cls, graph_id: str) -> autogpt_server.data.graph.Graph:
+        graph = await autogpt_server.data.graph.get_graph(graph_id, is_template=True)
         if not graph:
             raise HTTPException(status_code=404, detail=f"Graph #{graph_id} not found.")
         return graph
 
     @classmethod
-    async def create_new_graph(cls, graph: Graph) -> Graph:
+    async def create_new_template(
+        cls, create_graph: CreateGraph
+    ) -> autogpt_server.data.graph.Graph:
+        if create_graph.graph:
+            graph = create_graph.graph
+        elif create_graph.tempalte_id:
+            graph = await autogpt_server.data.graph.get_graph(
+                create_graph.tempalte_id, is_template=True
+            )
+        else:
+            raise HTTPException(
+                status_code=400, detail="Either graph or template_id must be provided."
+            )
+
+        # TODO: replace uuid generation here to DB generated uuids.
```

Comment:
duplicated logic moved this to shared method ? 

or can it be simpler: create_new_template = create_new_graph with is_template on the graph model = True

Reply:
Yeah, I'm still leaning toward having distinct endpoints though even if the backend is essentially the same

Reply:
afaics this has been fixed